src/features/marketplace/process/commands/change-price-item/store-change-price-item-command.ts

```diff
@@ -0,0 +1,33 @@
+import ItemService from '../../../../leda-nft/services/item.service';
+import MarketplaceError from '../../enums/marketplace-error.enum';
+import ICommand from '../../interfaces/command.interface';
+import MarketplaceState from '../../types/marketplace-state';
+
+export default class StoreChangePriceItemCommand implements ICommand<MarketplaceState> {
+  private readonly itemService: ItemService;
+
+  constructor(_itemService: ItemService) {
+    this.itemService = _itemService;
+  }
+
+  async execute(state: MarketplaceState): Promise<MarketplaceState> {
+    if (!state.itemId) return { ...state, error: MarketplaceError.RequiredItemId };
+    if (!state.price) return { ...state, error: MarketplaceError.RequiredPrice };
+    if (!state.marketplaceEvent) return { ...state, error: MarketplaceError.RequiredListEvent };
+
+    try {
+      state.item = await this.itemService.list(
+        state.itemId,
+        state.price,
+        state.listId,
+        state.ownerAddress
+      );
+    } catch (ex) {
+      // TODO: Handle exceptions properly
+      console.log('ex|StoreChangePriceItemCommand', ex);
+      return { ...state, error: MarketplaceError.StoreListItemFailure };
```

Comment:
You should throw the proper error here, StoreChangePriceFailure or something like that